ledger/participant-integration-api/src/main/resources/db/migration/oracle-appendonly/V100_0__Append_only_schema.sql

```diff
@@ -0,0 +1,484 @@
+
+-- filtering by template
+CREATE INDEX participant_events_divulgence_template_id_idx ON participant_events_divulgence(template_id);
+
+-- filtering by witnesses (visibility) for some queries used in the implementation of
+-- GetActiveContracts (flat), GetTransactions (flat) and GetTransactionTrees.
+-- Note that Potsgres has trouble using these indices effectively with our paged access.
+-- We might decide to drop them.
+CREATE INDEX participant_events_divulgence_tree_event_witnesses_idx ON participant_events_divulgence(JSON_ARRAY(tree_event_witnesses));
+
+-- lookup divulgance events, in order of ingestion
+CREATE INDEX participant_events_divulgence_contract_id_idx ON participant_events_divulgence(contract_id, event_sequential_id);
+
+
+---------------------------------------------------------------------------------------------------
+-- Events table: create
+---------------------------------------------------------------------------------------------------
+
+-- offset index: used to translate to sequential_id
+CREATE INDEX participant_events_create_event_offset ON participant_events_create(event_offset);
+
+-- sequential_id index for paging
+CREATE INDEX participant_events_create_event_sequential_id ON participant_events_create(event_sequential_id);
+
+-- lookup by event-id
+CREATE INDEX participant_events_create_event_id_idx ON participant_events_create(event_id);
+
+-- lookup by transaction id
+CREATE INDEX participant_events_create_transaction_id_idx ON participant_events_create(transaction_id);
+
+-- filtering by template
+CREATE INDEX participant_events_create_template_id_idx ON participant_events_create(template_id);
+
+-- filtering by witnesses (visibility) for some queries used in the implementation of
+-- GetActiveContracts (flat), GetTransactions (flat) and GetTransactionTrees.
+-- Note that Potsgres has trouble using these indices effectively with our paged access.
+-- We might decide to drop them.
+-- TODO these indices are never hit
+CREATE INDEX participant_events_create_flat_event_witnesses_idx ON participant_events_create(JSON_ARRAY(flat_event_witnesses));
+CREATE INDEX participant_events_create_tree_event_witnesses_idx ON participant_events_create(JSON_ARRAY(tree_event_witnesses));
+
+-- lookup by contract id
+-- TODO double-check how the HASH should work and that it is actually hit
+CREATE INDEX participant_events_create_contract_id_idx ON participant_events_create(ORA_HASH(contract_id));
+
+-- lookup by contract_key
+CREATE INDEX participant_events_create_create_key_hash_idx ON participant_events_create(create_key_hash, event_sequential_id);
+
+
+---------------------------------------------------------------------------------------------------
+-- Events table: consuming exercise
+---------------------------------------------------------------------------------------------------
+
+-- offset index: used to translate to sequential_id
```

Comment:
the index creation is separated originally from the tables to accommodate iterative migrations of heavy databases. since this is not needed, I would go ahead and put these next to the table definitions for clarity

Reply:
reordered to put indices right under their relevant table defs